client/src/ProvGraph.tsx

```diff
@@ -0,0 +1,102 @@
+import { Component } from 'react';
+import Location from './Models/Location';
+import {Dropdown, DropdownButton} from 'react-bootstrap';
+import { LineChart, Line, XAxis, YAxis, Label, Tooltip } from 'recharts';
+import { extent as d3Extent, max as d3Max } from 'd3-array';
+import { scaleLinear as d3ScaleLinear, scaleTime as d3ScaleTime, tickFormat} from 'd3-scale';
+import { format as d3Format } from 'd3-format';
+
+export default class ProvGraph extends Component<{longitude: any, latitude: any}, {prov: any, location: Location, selectedData: string}> {
+    async componentWillReceiveProps(nextProps: {latitude: any, longitude: any}): Promise<void> {
+        if (nextProps.latitude !== 0 && nextProps.longitude !== 0) {
+            this.setState({prov: await this.getCovidStatsByLocation(nextProps.latitude, nextProps.longitude)});
+            this.setState({location: await this.getLocationFromPosition(nextProps.latitude, nextProps.longitude)});
+        }
+    }
+
+    componentWillUnmount() {
+        this.setState = (state,callback)=>{
+            return;
+        };
+    }
+    
+    private getTime(date?: Date) {
+        return date != null ? date.getTime() : 0;
+    }
+
+    async getCovidStatsByLocation(latitude: number, longitude: number): Promise<any> {
+
+        try {
+            const url = `http://localhost:8080/location?longitude=${longitude.toPrecision(5)}&latitude=${latitude.toPrecision(5)}&component=provgraph`;
```

Comment:
As mentioned in provstats, this is not how rest API urls should be written. Needs to be reworked.